ui/lib/sync/addon/components/secrets/page/overview.ts

```diff
@@ -67,28 +65,12 @@ export default class SyncSecretsDestinationsPageComponent extends Component<Args
   });
 
   @action
-  resetOptInModal() {
-    this.showActivateSecretsSyncModal = false;
-    this.hasConfirmedDocs = false;
+  onModalError(errorMsg: string) {
+    this.activationError = errorMsg;
   }
 
-  @task
-  @waitFor
-  *onFeatureConfirm() {
-    // must return null instead of root for non managed cluster.
-    // child namespaces are not sent.
-    const namespace = this.args.isHvdManaged ? 'admin' : null;
-    try {
-      yield this.store
-        .adapterFor('application')
-        .ajax('/v1/sys/activation-flags/secrets-sync/activate', 'POST', { namespace });
-      // must refresh and not transition because transition does not refresh the model from within a namespace
-      yield this.router.refresh();
-    } catch (error) {
-      this.error = errorMessage(error);
-      this.flashMessages.danger(`Error enabling feature \n ${errorMessage(error)}`);
-    } finally {
-      this.resetOptInModal();
-    }
```

Comment:
this code is all still here, i just moved it into the modal component 🎉

https://github.com/hashicorp/vault/pull/26822/files#diff-90605331528edf71f7c228875e939fd17825f4edd217b7c1563a1d88cc23763eR30-R50